MIGRATION_GUIDE.md

```diff
@@ -127,4 +129,28 @@ api.Client.Create()
 ```
 </td>
 </tr>
-</table>
\ No newline at end of file
+</table>
+
+### Changes To Retry Strategy
+
+Previously, when retrying requests the SDK would retry only for 429 status codes, waiting until the time indicated by the `X-RateLimit-Reset` header and would not have a maximum number of requests (i.e would retry infinitely).
+
+Now, by default the SDK will retry 429 status codes and request errors that are deemed to be recoverable, and will retry a maximum of 2 times. Instead of waiting for the `X-RateLimit-Reset` header value, it will instead use an exponential backoff with a minimum time of 250 milliseconds.
+
+This logic can be customised by using the `management.WithRetries` helper, and can be disabled using the `management.WithNoRetries`
+
+```go
+// Enable a retry strategy with 3 retries that will retry on 429 and 503 status codes
+m, err := management.New(
+    domain,
+    management.WithClientCredentials(context.Background(), id, secret),
+    management.WithRetries(RetryOptions{MaxRetries: 3, Statuses: []int{http.StatusTooManyRequests, http.StatusBadGateway}}),
+)
+
+// Disable retries
+m, err := management.New(
+    domain,
+    management.WithClientCredentials(context.Background(), id, secret),
+    management.WithNoRetries(),
```

Comment:
I understand it might be a bit late for this, but with this design it would be possible to use both `WithRetries()` and `WithNoRetries()` –even though the last used method would prevail. What about simply using `management.WithRetries(RetryOptions{MaxRetries: 0})` ?

Reply:
Consider as well using factory methods if necessary, to reduce the verbosity.

Reply:
Also, what is the SDK already doing with other similar configuration options?

Reply:
Yeah I was a little undecided on this and initially had an `Enabled` property on the `RetryOptions` struct.

We currently have `WithNoAuth0ClientInfo` to disable the Auth0Client telemetry so that was part of the reason in going for `WithNoRetries` for this API

Reply:
Note that this being a new major version, it's an opportunity to go with the approach that makes the most sense for all methods involved (including `WithNoAuth0ClientInfo`). It's alright if you want to use `WithNoRetries()` anyway though.

Reply:
I think given that we'd most likely have keep `WithNoAuth0ClientInfo` (i.e. adding an `Enabled` flag to that struct isn't as straight forward), then I'd lean towards maintaining the existing `WithX`/`WithNoX` pattern as it does fit with Go